subprojects/build-cache/src/main/java/org/gradle/caching/internal/origin/OriginMetadataFactory.java

```diff
@@ -40,77 +35,67 @@
         this.userName = userName;
         this.operatingSystem = operatingSystem;
-        this.clock = clock;
-        this.gradleVersion = gradleVersion;
+        this.additionalProperties = additionalProperties;
         this.currentBuildInvocationId = currentBuildInvocationId;
     }
 
     public OriginWriter createWriter(CacheableEntity entry, long elapsedTime) {
         return new OriginWriter() {
             @Override
-            public void execute(OutputStream outputStream) {
-                // TODO: Replace this with something better
+            public void execute(OutputStream outputStream) throws IOException {
                 Properties properties = new Properties();
-                properties.setProperty(BUILD_INVOCATION_ID_KEY, currentBuildInvocationId.asString());
+                properties.setProperty(BUILD_INVOCATION_ID_KEY, currentBuildInvocationId);
                 properties.setProperty(TYPE_KEY, entry.getClass().getCanonicalName());
                 properties.setProperty(IDENTITY_KEY, entry.getIdentity());
-                properties.setProperty(GRADLE_VERSION_KEY, gradleVersion.getVersion());
-                properties.setProperty(CREATION_TIME_KEY, Long.toString(clock.getCurrentTime()));
+                properties.setProperty(CREATION_TIME_KEY, Long.toString(System.currentTimeMillis()));
                 properties.setProperty(EXECUTION_TIME_KEY, Long.toString(elapsedTime));
                 properties.setProperty(ROOT_PATH_KEY, rootDir.getAbsolutePath());
                 properties.setProperty(OPERATING_SYSTEM_KEY, operatingSystem);
                 properties.setProperty(HOST_NAME_KEY, inetAddressFactory.getHostname());
                 properties.setProperty(USER_NAME_KEY, userName);
-                try {
-                    properties.store(outputStream, "Generated origin information");
-                } catch (IOException e) {
-                    throw UncheckedException.throwAsUncheckedException(e);
-                }
-                assert METADATA_KEYS.containsAll(properties.stringPropertyNames()) : "Update expected metadata property list";
+                additionalProperties.accept(properties);
+                properties.store(outputStream, "Generated origin information");
             }
         };
     }
 
     public OriginReader createReader(CacheableEntity entry) {
         return new OriginReader() {
             @Override
-            public OriginMetadata execute(InputStream inputStream) {
-                // TODO: Replace this with something better
+            public OriginMetadata execute(InputStream inputStream) throws IOException {
                 Properties properties = new Properties();
-                try {
-                    properties.load(inputStream);
-                } catch (IOException e) {
-                    throw UncheckedException.throwAsUncheckedException(e);
+                properties.load(inputStream);
+                if (LOGGER.isDebugEnabled()) {
```

Comment:
What is the reason for guarding this?
I see that most `debug` usages don't contain a guard, and the parameter creations seem cheap.

Reply:
The `getDisplayName()`.